src/main/scala/com/fulcrumgenomics/umi/ExtractUmisFromBam.scala

```diff
@@ -161,7 +170,7 @@ class ExtractUmisFromBam
           r1.name = r1.name + "+" + bases1 + bases2
           r2.name = r2.name + "+" + bases1 + bases2
         }
-        assert(r1.name.equals(r2.name))
+        assert(r1.name.equals(r2.name), s"Mismatching read name.  R1: '$r1' R2: '$r2'")
```

Comment:
Just my 2c, but I feel like `assert` is for cases where you are asserting something that should _always_ be true and a failure indicates a bug, vs. `require` for cases where you are requiring a condition that could be made to fail by being given incorrect input.